src/com/amazon/kinesis/streaming/agent/tailing/AbstractParser.java

```diff
@@ -460,11 +492,21 @@ private R buildRecord(int offset, int length) {
         ByteBuffer data = ByteBuffers.getPartialView(currentBuffer, offset, length);
         ++recordsFromCurrentBuffer;
         Preconditions.checkNotNull(currentBufferFile);
-        R record = buildRecord(currentBufferFile, data, toChannelOffset(offset));
-        totalRecordsParsed.incrementAndGet();
+        R record = null;
+        try {
+            record = buildRecord(currentBufferFile, convertData(data), toChannelOffset(offset));
+            totalRecordsParsed.incrementAndGet();
+        } catch (DataConversionException e) {
```

Comment:
fair enough. Actually it would be nice to have some configurable fallback policy, but that requires time to implement :)